src/MUIDataTable.js

```diff
@@ -535,13 +536,17 @@ class MUIDataTable extends React.Component {
   }
 
   sortTable(data, col, order) {
-    let sortedData = data.map((row, index) => ({
+    let dataSrc = this.options.customSort ? this.options.customSort(data, col, order || "asc") : data;
+
+    let sortedData = dataSrc.map((row, sIndex) => ({
       data: row[col],
-      position: index,
-      rowSelected: this.state.selectedRows.indexOf(index) >= 0 ? true : false,
+      position: sIndex,
+      rowSelected: this.state.selectedRows.indexOf(sIndex) >= 0 ? true : false,
```

Comment:
The ternary operator is not needed right? (I'm blaming the original commit :p)
```suggestion
      rowSelected: this.state.selectedRows.indexOf(sIndex) >= 0,
```